packages/video_player/video_player_android/android/src/main/java/io/flutter/plugins/videoplayer/VideoCache.java

```diff
@@ -0,0 +1,59 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.videoplayer;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.android.exoplayer2.database.DatabaseProvider;
+import com.google.android.exoplayer2.database.StandaloneDatabaseProvider;
+import com.google.android.exoplayer2.upstream.cache.LeastRecentlyUsedCacheEvictor;
+import com.google.android.exoplayer2.upstream.cache.SimpleCache;
+import java.io.File;
+
+public class VideoCache {
+  private static final String cacheFolder = "exoCache";
+  private static SimpleCache sDownloadCache;
+
+  @NonNull
+  public static SimpleCache getInstance(@NonNull Context context, long maxCacheSize) {
+    DatabaseProvider databaseProvider = new StandaloneDatabaseProvider(context);
+
+    if (sDownloadCache == null)
+      sDownloadCache =
+          new SimpleCache(
+              new File(context.getCacheDir(), cacheFolder),
+              new LeastRecentlyUsedCacheEvictor(maxCacheSize),
+              databaseProvider);
+    return sDownloadCache;
+  }
+
+  public static void clearVideoCache(@NonNull Context context) {
+    try {
+      File dir = new File(context.getCacheDir(), cacheFolder);
+      deleteDir(dir);
```

Comment:
If deleting fails, should we let the developer know or throw an exception?

Reply:
@camsim99 Let me discuss this with the team. I absolutely agree that informing the developer is important. But I'm not sure how to implement this at the moment. Brb.

Reply:
Oke, I think I'll implement some sort of ClearCacheMessageResponse with a boolean (hasSucceed). In future, we could add some additional information to the ClearCacheMessageResponse. e.g. error messages or additional instructions. Agree?

Reply:
That makes sense to me because the boolean would allow to error to be handled on the Dart side

Reply:
Done